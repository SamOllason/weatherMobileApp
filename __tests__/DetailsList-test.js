/**
 * @format
 */

import 'react-native';
import React from 'react';
import DetailsList from '../DetailsList';
import { shallow } from 'enzyme';

// Using Jest + Enzyme
describe('<DetailsList />', () => {
    it('renders correctly, testing using Jest+Enzyme', () => {
        expect.assertions(1);

        const mockForecast = [ {
            "dt": 1534852800,
            "main": {
                "temp": 290.19,
                "temp_min": 290.19,
                "temp_max": 291.928,
                "pressure": 999.46,
                "sea_level": 1018.59,
                "grnd_level": 999.46,
                "humidity": 97,
                "temp_kf": -1.74
            },
            "weather": [
                {
                    "id": 501,
                    "main": "Rain",
                    "description": "中雨",
                    "icon": "10d"
                }
            ],
            "clouds": {
                "all": 92
            },
            "wind": {
                "speed": 5.01,
                "deg": 252.501
            },
            "rain": {
                "3h": 8.56
            },
            "sys": {
                "pod": "d"
            },
            "dt_txt": "2018-08-21 12:00:00"
        }];

        const mockCurrentWeather = "cloudy";

        expect(shallow(<DetailsList forecast={mockForecast} currentWeather={mockCurrentWeather}/>)).toMatchSnapshot();
    });
});