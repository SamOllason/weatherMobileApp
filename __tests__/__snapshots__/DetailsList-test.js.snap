// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsList /> renders correctly, testing using Jest+Enzyme 1`] = `
<ScrollViewMock>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      key="2018-08-21 12:00:00"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "margin": 1,
          "padding": 10,
          "paddingLeft": 30,
          "paddingRight": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "whitesmoke",
              "fontSize": 15,
            }
          }
        >
          Tue 12pm
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "whitesmoke",
              "fontSize": 15,
            }
          }
        >
          Rain
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "whitesmoke",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
        >
          291.928
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "whitesmoke",
              "fontSize": 15,
            }
          }
        >
          290.19
        </Text>
      </View>
    </View>
  </View>
</ScrollViewMock>
`;
